### DISCUSSION TITLE
Problem on auth - 403

### DISCUSSION THREAD

**Discussion by labdel** [timdorr/tesla-api#621] [Q&A]
My python program is broken now. I a mgettin 403 in step 2. I noticed the TESLA login form "https://auth.tesla.com/oauth2/v3/authorize" comes in 2 steps: First it asks for the e-mail address, once button NEXT is hit it redraws the page and adds the Password input. Upon a successfull login it takes to the page https://auth.tesla.com/void/callback where We pull the "Code".

Has anyone resolved this ? I mean the fact that there are 2 submissions now. I am followin up at https://tesla-api.timdorr.com/api-basics/authentication#post-https-auth.tesla.com-oauth2-v3-token-1 but it seems I am missing something.

I would appreciate any guidance.

Thanks in advance ! - Leo.

**Comment by amirhmk** [comment]
Yep seems like the [authentication page](https://auth.tesla.com/oauth2/v3/authorize?client_id=ownerapi&code_challenge=ZTdhNWQwODhmNTkwZGVmNjQwMzk4MzY3NmI1NTAzODA3YTVkMzQ1YzhkNTZiYTYxYTNlNjA1NGFhMGNmZDg3ZGI4YzQzMjc0ZTMwNWJhNWI2ZDc5MjI&code_challenge_method=S256&redirect_uri=https%3A%2F%2Fauth.tesla.com%2Fvoid%2Fcallback&response_type=code&scope=openid+email+offline_access&state=MTY3MzUwZDM4ZmQ5) has two steps now...

**Reply by amirhmk** [reply]
@labdel actually this should work. Make sure you're setting your headers correctly, that was my issue. 

```
headers = {
            "User-Agent": 
            "x-tesla-user-agent":
            "X-Requested-With": "com.teslamotors.tesla",
}
```
